### PR TITLE
Fix potential problem with binary_search_by_key()

### LINKED ISSUE #1162: CLI: Investigate whether `Vec<AddressWithUnspentOutputs` is sorted before a binary search (potential bug in the CLI)
## Description

While implementing a feature I noticed that we do a binary-search on a `Vec<AddressWithUnspentOutputs>` as part of the `addresses` command in the CLI. For this to work the vec needs to be in order before doing a search, but that might not be the case.

## Motivation

Potential bug.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Fix potential problem with binary_search_by_key()

## Links to any relevant issues

Closes https://github.com/iotaledger/iota-sdk/issues/1162

## Type of change

- Enhancement

## How the change has been tested

Running the addresses command

### CODE REVIEW

**Review by DaughterOfMars — DISMISSED** [review]

**Review by thibault-martinez — COMMENTED** [review]
If the bug was really existing, it's probably worth of a changelog entry

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 1.0.1 - 2023-MM-DD
```
Comment (thibault-martinez) [inline cli/CHANGELOG.md:22]: Tbh if we have 1.1 very soon, maybe I'd just do it there?

Comment (Thoralf-M) [reply cli/CHANGELOG.md:22]: Let's just change when we know the date

Comment (thibault-martinez) [reply cli/CHANGELOG.md:22]: Yeah but I mean, do we want to target the 1.1 branch then ?

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**